src/components/AppIndex.vue

```diff
@@ -18,6 +18,25 @@ export default {
             recipes: []
         }
     },
+    computed: {
+        /**
+         * Is the Cookbook recipe directory currently being changed?
+         */
+        updatingRecipeDirectory () {
+            return this.$store.state.updatingRecipeDirectory
+        }
+    },
+    watch: {
+        /**
+         * If the Cookbook recipe directory currently was changed, reload
+         * the recipes in the index component.
+         */
+        updatingRecipeDirectory(newVal, oldVal) {
+            if (newVal == false && newVal != oldVal) {
+                this.loadAll()
```

Comment:
This will (currently) only load the new recipes. The categories on the left side will not get updated.

Reply:
Yes, this only updates this component (itself). The categories are updated [here](https://github.com/nextcloud/cookbook/blob/b7ac727a100653d4c11ddcca0723a89ab43bb7f0/src/components/AppNavi.vue#L300)

Reply:
I do not know by hard, where the problem is exactly but [here](https://github.com/nextcloud/cookbook/blob/b7ac727a100653d4c11ddcca0723a89ab43bb7f0/src/components/AppNavi.vue#L275) I get the error that `$this.$refs['app-navi-cat-'+i][0]` is undefined when switching back and forth.
Looking at the `$this.$refs` using the JS debugger + Vue dev tools indicated that some of the `app-navi-cat-*` entries are empty. Seems like there is something not synchronized on time.

I have not elaborated further, just saw the error in the JS console and had a quick glance on why this is happening.

Reply:
I’ll have a look at this. With "switching back and forth" you mean switching between different recipe directories, I guess?

Reply:
The problem was that the component tried loading recipes for the categories before they were updated in by vue. This was actually already a problem before but couldn’t be notice since updating the recipe folder "failed" anyway ;)
I updated the PR to fix this